Lib/test/test_functools.py

```diff
@@ -249,6 +256,16 @@ def test_placeholders_optimization(self):
         self.assertEqual(p2.args, (PH, 0))
         self.assertEqual(p2(1), ((1, 0), {}))
 
+    def test_placeholders_kw_restriction(self):
+        PH = self.module.Placeholder
+        with self.assertRaisesRegex(TypeError, "Placeholder"):
+            self.partial(capture, a=PH)
+        # Passes, as checks via `is` and not `eq`
+        p = self.partial(capture, a=unittest.mock.ANY)
+        actual_args, actual_kwds = p()
+        self.assertEqual(actual_args, ())
+        self.assertEqual(actual_kwds, {'a': unittest.mock.ANY})
```

Comment:
You can add the same "is" test here. It is not necessary, as the returned values does not really matter. In any case I already approved this PR and just wait a while for the case other core devs have a word.

BTW, you could also use `test.support.ALWAYS_EQ`, which is similar to `unittest.mock.ANY` but has more narrow purpose. Both `unittest.mock` and `test.support` are already imported. It is up to you what to prefer.